### PR TITLE
[Enhance] Add support for dict input in Choice

### PR DESCRIPTION
**Pull request by KKIEEK**
### Usage
```python
some = dict(
    type='Choice',
    categories=dict(
        alias1=complicated_obj1,
        alias2=complicated_obj2))

# It is equivalent to the above
some = dict(
    type='Choice',
    categories=[complicated_obj1, complicated_obj2],
    alias=['alias1', 'alias2']))
```

### CODE REVIEW

File: siatune/tune/spaces/choice.py [hunk 1]
```diff
@@ -14,16 +14,20 @@ class Choice(BaseSpace):
...
-            Defaults to None.
+        categories (Sequence | dict): The categories. If categories is dict,
+            keys of dict will overwrite the alias.
+        alias (Sequence, optional): A alias to be expressed. Defaults to None.
     """
 
     sample: Callable = tune.choice
 
     def __init__(self,
                  categories: Sequence,
```
Comment (yhna940) [inline siatune/tune/spaces/choice.py:25]:
```suggestion
                 categories: Union[Sequence, Dict] ,
```

File: siatune/tune/spaces/choice.py [hunk 2]
```diff
@@ -14,16 +14,20 @@ class Choice(BaseSpace):
     """Sample a categorical value.
 
     Args:
-        categories (Sequence): The categories.
-        alias (Sequence, optional): A alias to be expressed.
-            Defaults to None.
+        categories (Sequence | dict): The categories. If categories is dict,
```
Comment (nijkah) [inline siatune/tune/spaces/choice.py:17]: How about describing it as 'The categorical search space to choice one'?

File: siatune/tune/spaces/choice.py [hunk 3]
```diff
@@ -14,16 +14,20 @@ class Choice(BaseSpace):
...
+            keys of dict will overwrite the alias.
+        alias (Sequence, optional): A alias to be expressed. Defaults to None.
```
Comment (nijkah) [inline siatune/tune/spaces/choice.py:19]:
```suggestion
        alias (Sequence, optional): An alias to be expressed. Defaults to None.
```

**Review by yhna940 — APPROVED** [review]

File: siatune/tune/spaces/choice.py (hunk 2 above)
Comment (KKIEEK) [reply siatune/tune/spaces/choice.py]: I got it.

**Review by nijkah — APPROVED** [review]

**Review by nijkah — APPROVED** [review]

**Pull request merged**